### DISCUSSION TITLE
AzerothCore License Transition Announcement

### DISCUSSION THREAD

**Discussion by Nyeriah** [azerothcore/azerothcore-wotlk#19868] [General]
Dear AzerothCore Community,

We hope this message finds you well. We are writing to address an important update regarding the licensing of AzerothCore and to provide clarity on the changes made in the past.

**AzerothCore’s History and Previous Licensing**

AzerothCore has evolved through the dedicated efforts of numerous contributors and projects over the years.  The project initially began based on the source code of Sunwellcore, published by its original authors, Xinef and Pussywizard (you can see the original Sunwellcore release notes at this page https://www.azerothcore.org/pages/sunwell.pl/ ).

Sunwellcore itself was based on a 2012 version of TrinityCore and had been developed privately before its public release. When Sunwellcore was made available to the public, its source code was released without any associated git history or license. TrinityCore, like its predecessor MaNGOS, was licensed under GPLv2+, which required any derivative work to carry the same license.

When we released AzerothCore, we applied the AGPL license to the project. Our intention was to prevent individuals from using AzerothCore without contributing back, in line with the collaborative spirit of our open-source philosophy. However, this was a mistake on our part. Technically, it was not possible to apply the AGPL license to Sunwellcore because, despite its release without a license, its code was based on TrinityCore/MaNGOS and therefore must be considered as falling under the GPLv2+ license.

We fully acknowledge this oversight and understand the significant concerns and grievances it has caused within the emulation community. We deeply regret any inconvenience this may have caused and are committed to resolving this issue permanently.

We want to emphasize that we have always been dedicated to promoting cross-collaboration among open-source projects within the WoW emulation community.

**License Change and Correction**

As mentioned, there was a change in licensing that diverged from our historical precedent. This change was initially intended to address specific needs and considerations at the time; however, after careful reflection and discussions within the community, it became evident that reverting to the original licensing terms would better align with our project's foundational principles and the expectations of our contributors.

We believed that transitioning to AGPL3 was the right move because it supported our goal of fostering a collaborative environment where every contributor plays a role in creating a bug-free emulator. We envisioned AGPL3 as a means to ensure that contributions and improvements were shared transparently and that the project benefited from collective input. However, we now recognize that this approach was not feasible under our current constraints and that reverting to our original GPL2+ license is more in keeping with the values and historical context of AzerothCore.

This decision is driven by our commitment to maintaining the openness and inclusivity that has always been at the heart of AzerothCore. By returning to GPL2+, we are reinforcing our dedication to providing a platform where innovation and shared knowledge can continue to flourish.

**What This Means for the Community**

Reverting from AGPL3 to GPL2+ represents a significant shift for the AzerothCore community. By moving back to GPL2+, we are reaffirming our commitment to an open and collaborative environment. This change ensures that all contributions and modifications remain freely accessible, preserving the transparency that has always been a cornerstone of our project.

In practice, this means that our license will now be compatible with the largest emulation projects, such as TrinityCore and other MaNGOS-based projects, which should allow for better collaboration between our projects.

We understand that changes in licensing can have significant implications for our contributors and users. Therefore, we want to assure you that this transition has been carefully considered and reflects our commitment to the values that have always guided AzerothCore.

**Acknowledgments and Moving Forward**

We extend our heartfelt thanks to all contributors, developers, and community members who have supported AzerothCore throughout its journey. Your dedication and passion are the driving forces behind our continued success.

As we move forward with this license transition, we invite you to engage with us, share your feedback, and continue to be an integral part of the AzerothCore community. Together, we will continue to build and enhance this remarkable project.

If you have any questions or need further clarification regarding the licensing change, please feel free to reach out to us via our official channels.

If you have contributed to AzerothCore at any time in the past five years, please leave a message >[HERE](https://github.com/azerothcore/azerothcore-wotlk/discussions/19869)< with your explicit consent on the commits you authored being relicensed to the GPLv2+ license.

Thank you for your ongoing support and for being an essential part of the AzerothCore legacy.

Warm regards,

The AzerothCore Team

**Comment by TesterWoWDev** [comment]
Linking for further information, as this was discussed roughly 2 years ago: https://github.com/ihm-tswow/azerothcore-open-letter#happening-1-h1-removing-gplv2-or-later-headers

**Comment by ihm-tswow** [comment]
This could be more clear for what it means that your code to have become "compatible with other emulators".

> Technically, it was not possible to apply the AGPL license to Sunwellcore because, despite its release without a license, its code was based on TrinityCore/MaNGOS and therefore must be considered as falling under the GPLv2+ license

It might be your interpretation that sunwell intended for their contributions to be used under the GPLv2 or even more permissive terms, and I have little reason to contest that (as far as i recall the wording was very permissive), but your assertion is not correct. Releasing code without any clear license is not an automatic pass to consider it under the license of the work it was based on, even when the license itself demands it. Such a breach would be between the person who released the code and the people whose work they derived it from and in regards to the derived code.

That said, at this point I'd probably happily accept sunwell changes given a review of their statement, seeing how long we've been able to operate under it, but I cannot speak for other projects on this.

However, the same cannot be said of individual contributions made clearly under the AGPL for work not entirely derived from TC/MaNGOS/Sunwell and by people on your relicensing consent list. For those looking to port azerothcore patches to other emulators, it is very important that you continue to take care to declare any non-trivial code not consensually relicensed. I would not consider such contributions as made by azerothcore as a whole, but by the individuals that wrote it and terms they agreed to share it on, and I doubt other projects would think differently.

That said, happy to see this probably means future relicensing issues are less likely to occur, and wish you the best of luck with your project going forward.

**Comment by ZaDarkSide** [comment]
When change?

**Comment by brndd** [comment]
Hey. Hate to play Clippy here as an outsider, but I was looking at the OK's in your consent thread and noticed quite a few are still missing, including the second-most prolific contributor since Jan. 1, 2020. Have you considered pinging all as-of-yet unconsenting contributors on GitHub to reach them? This is what other open source projects I have contributed to that have relicensed have done.
